packages/maskbook/src/plugins/Wallet/services/wallet.ts

```diff
@@ -95,7 +95,31 @@ export async function updateExoticWalletFromSource(
     }
     for (const address of updates.keys()) {
         const wallet = await walletStore.get(formatChecksumAddress(address))
-        if (wallet) continue
+        let update = false
+        if (wallet) {
+            [
+                'erc20_token_blacklist',
+                'erc20_token_whitelist',
+                'erc721_token_blacklist',
+                'erc721_token_whitelist',
+                'erc1155_token_blacklist',
+                'erc1155_token_whitelist'
+            ].forEach(x => {
```

Comment:
```ts
function setWallet(key: keyof typeof wallet) {
  if (!Reflect.has(wallet, x)) {
    Reflect.set(wallet, x, new Set())
    update = true
  }
}

setWallet('erc20_token_blacklist')
setWallet('erc20_token_whitelist')
setWallet('erc721_token_blacklist')
setWallet('erc721_token_whitelist')
setWallet('erc1155_token_blacklist')
setWallet('erc1155_token_whitelist')
```

Reply:
Unluckily declare function inside a for loop is bad practice.

Reply:
Is this possbile?

```js
// Declare a function outside of loop
function setWallet(key: keyof WalletRecordInDatabase, value: WalletRecordInDatabase) {
    if (!Reflect.has(value, key)) {
        Reflect.set(value, key, new Set())
        return true
    }
    return false
}


update = [setWallet('erc20_token_blacklist', wallet),
          setWallet('erc20_token_whitelist', wallet),
          setWallet('erc721_token_blacklist', wallet),
          setWallet('erc721_token_whitelist', wallet),
          setWallet('erc1155_token_blacklist', wallet),
          setWallet('erc1155_token_whitelist', wallet)].some((item) => item)
```

Reply:
not readable at a glance, and still adding an another array..